### ISSUE TITLE
[BUG][Dart] dioNext: BuiltList<String> gets brackets in query string

### ISSUE THREAD

**Issue by markbeij** [OpenAPITools/openapi-generator#9522]
##### openapi-generator version
openapi_generator: ^3.1.3

##### OpenAPI declaration file content or url
https://petstore.swagger.io/#/pet/findPetsByStatus

##### Generation Details

`flutter pub run build_runner build --delete-conflicting-outputs`

```
@Openapi(
    additionalProperties: AdditionalProperties(pubName: 'petstore_api_client', pubAuthor: 'www.blah.com'),
    inputSpecFile: 'https://petstore.swagger.io/v2/swagger.json',
    generatorName: Generator.dioNext,
    outputDirectory: '../petstore',
    overwriteExistingFiles: true,
    alwaysRun: true)
class ShuttelApiClientGeneratorConfig extends OpenapiGeneratorConfig {}
```

##### Steps to reproduce
1. Generate client using `Generator.dioNext`
1. Execute unit test: https://gist.github.com/markbeij/c23b32d4dbda7477ff673f2b3f9e827a.js

Expected url is: https://petstore.swagger.io/v2/pet/findByStatus?status=available
But got: https://petstore.swagger.io/v2/pet/findByStatus?status=%5Bavailable%5D

**Comment by markbeij** [comment]
@kuhnroyal maybe you can take a look at this?

**Comment by markbeij** [comment]
Thats a great question and I think it differs per API.
The petstore api expects https://petstore.swagger.io/v2/pet/findByStatus?status=available&status=pending&status=sold

**Comment by kuhnroyal** [comment]
Ok, this should be possible once I get around to solve #6681

**Comment by kuhnroyal** [comment]
Should be fixed by #9635 - would be great if you could test.
